### PR TITLE
CBAPI-4818: Version number updates for Release 1.5.0

### PR DESCRIPTION
**Pull request by abowersox-cb**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Tests have been added that prove the fix is effective or that the feature works.
- [ ] New and existing tests pass locally with the changes.
- [x] Code follows the style guidelines of this project (PEP8, clean code, linter).
- [x] A self-review of the code has been done.

## What is the ticket or issue number?

https://jira.carbonblack.local/browse/CBAPI-4818

## Pull Request Description

This is Step 4 of the Release Process. Version numbers have been updated and a changelog has been created.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## How Has This Been Tested?

Changelog has been compiled and viewed locally to validate the formatting, especially the internal link to the Alert Migration Guide.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # VMware Carbon Black Cloud Python SDK
 
-**Latest Version:** 1.4.3
+**Latest Version:** 1.5.0
 <br>
-**Release Date:** June 26, 2023
+**Release Date:** TBD
```
Comment (avanbrunt-cb) [inline README.md:5]: We should aim for today so when you need too update the date for today

Comment (abowersox-cb) [reply README.md:5]: That's in Step 1 of the "Releasing the SDK" process.

File: docs/changelog.rst [hunk 2]
```diff
@@ -1,5 +1,53 @@
 Changelog
 ================================
+CBC SDK 1.5.0 - Released (TBD)
+--------------------------------------
+
+**Breaking Changes:**
+
+* Alerts V7: This is a new version of the Alerts API with certain changes that are not compatible with code written
+  to the old V6 API. For details, please see the :ref:`Alert Migration Guide <alert-migration-guide>`.
```
Comment (kebringer-cb) [inline docs/changelog.rst:9]: link syntax needs a tweak. replace
:ref:`Alert Migration Guide <alert-migration-guide>`
with
`Alert Migration Guide <alert-migration-guide>`_

File: docs/changelog.rst [hunk 3]
```diff
@@ -1,5 +1,53 @@
...
+  * Additional fields available for both searching and faceting
+  * Enhanced note management with the ability to add notes to both individual alerts and threats (alerts grouped
+    by threat)
+  * Observed Alerts have been removed from the Alerts API as these events are not considered actionable threats. They
+    can now be retrieved via the Observations API.
+
+* Official support for Python 3.7 has been dropped, since that version is now end-of-life.  Added explicit testing
+  support for Python version 3.12.  **N.B.:** End users should update their Python version to 3.8.x or greater.
+
+New Features:
```
Comment (kebringer-cb) [inline docs/changelog.rst:26]: Should this be bold to match the breaking changes? Also Updates, Documentation

File: docs/changelog.rst [hunk 4]
```diff
@@ -1,5 +1,53 @@
...
+
+**Breaking Changes:**
+
+* Alerts V7: This is a new version of the Alerts API with certain changes that are not compatible with code written
+  to the old V6 API. For details, please see the :ref:`Alert Migration Guide <alert-migration-guide>`.
+  New features in Alerts include:
+
+  * Extended alert schema with additional metadata such as process command line and username, parent and child process
+    information, netconn data, additional device fields, MITRE categorization when available, and more
+  * Easier management and consumption of grouped alerts
```
Comment (avanbrunt-cb) [inline docs/changelog.rst:14]: This did not come in 1.5.0 it will be 1.5.1

File: docs/changelog.rst [hunk 5]
```diff
@@ -1,5 +1,53 @@
...
+* Process search validation has been changed to use the V2 ``POST`` API rather than the old V1 ``GET`` API.
+* ``CBCloudAPI.get_notifications()`` and ``CBCloudAPI.notification_listener()`` have been marked as deprecated.
+
+Documentation:
+
+* Added example script to poll for audit logs.
+* ``CBCloudAPI`` documentation has been pulled out into its own page.
+* Authentication, Getting Started, and Guides pages have been updated.
+* Concepts page has been removed, and the information it contained has moved to other pages.
+* New Searching guide added.
```
Comment (kebringer-cb) [inline docs/changelog.rst:43]: A link to the search guide would be cool.
https://carbon-black-cloud-python-sdk.readthedocs.io/en/latest/searching/

File: docs/changelog.rst [hunk 6]
```diff
@@ -1,5 +1,53 @@
 Changelog
 ================================
+CBC SDK 1.5.0 - Released (TBD)
+--------------------------------------
+
+**Breaking Changes:**
+
+* Alerts V7: This is a new version of the Alerts API with certain changes that are not compatible with code written
+  to the old V6 API. For details, please see the :ref:`Alert Migration Guide <alert-migration-guide>`.
+  New features in Alerts include:
```
Comment (kebringer-cb) [inline docs/changelog.rst:10]: Extra newline to separate New features...

File: docs/changelog.rst [hunk 7]
```diff
@@ -1,5 +1,53 @@
...
+
+  * Extended alert schema with additional metadata such as process command line and username, parent and child process
+    information, netconn data, additional device fields, MITRE categorization when available, and more
+  * Easier management and consumption of grouped alerts
+  * Ability to mark alerts as “In Progress”
+  * Ability to mark alerts as True Positive or False Positive
+  * Additional fields available for both searching and faceting
```
Comment (avanbrunt-cb) [inline docs/changelog.rst:17]: This doesn't seem like a breaking change

File: docs/changelog.rst [hunk 8]
```diff
@@ -1,5 +1,53 @@
 Changelog
 ================================
+CBC SDK 1.5.0 - Released (TBD)
+--------------------------------------
+
+**Breaking Changes:**
```
Comment (avanbrunt-cb) [inline docs/changelog.rst:6]: Based on the content below do we want make this section the summary and more explicitly call out the breaking changes

File: docs/changelog.rst (hunk 2 above)
Comment (abowersox-cb) [reply docs/changelog.rst]: This is link syntax for RST, not Markdown.

File: docs/changelog.rst (hunk 3 above)
Comment (abowersox-cb) [reply docs/changelog.rst:42]: They've never been so in the past.

**Review by kebringer-cb — APPROVED** [review]

**Pull request merged**